### PR TITLE
New collector for the stats of the versions script

### PR DESCRIPTION
**Pull request by seb128**

### CODE REVIEW

**Comment by seb128** [comment]
Sorry, I ended up submitting it again while trying to editing, I closed the previous requests now

File: metrics/collectors/versions/versions_collector.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+
+class VersionsMetrics(Metric):
+    def collect(self):
+        """ Collect the sponsoring queue details"""
+        data = []
+
+        known_reports_lst = []
+        with urllib.request.urlopen(VERSIONS_STATS_URL + "reports") as reports:
```
Comment (iainlane) [inline metrics/collectors/versions/versions_collector.py:19]: nitpick: can we use [`urljoin`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) to construct the URL please?

File: metrics/collectors/versions/versions_collector.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            for report in reports.read().decode("utf-8", errors="ignore").split("\n"):
```
Comment (iainlane) [inline metrics/collectors/versions/versions_collector.py:20]: semi nitpick:

```suggestion
            for report in reports:
                report = report.decode("utf-8", errors="ignore").strip()
                self.log.debug(f"found {report}")
```

(saves loading the whole thing at once)

File: metrics/collectors/versions/versions_collector.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                if report:
+                    known_reports_lst.append(report)
+
+        for report in known_reports_lst:
+            with urllib.request.urlopen(VERSIONS_STATS_URL + report) as stats:
+                for category in (
+                    stats.read().decode("utf-8", errors="ignore").split("\n")
```
Comment (iainlane) [inline metrics/collectors/versions/versions_collector.py:27]: as above

**Comment by iainlane** [comment]
Thanks! Some smallish nitpicks.

Question: I imagine/hope you'll want to add SRU version tracking to this eventually. Should we have a tag for the release right now? Maybe also a `devel` bool too.

**Review by iainlane — CHANGES_REQUESTED** [review]

File: metrics/collectors/versions/versions_collector.py (hunk 2 above)
Comment (seb128) [reply metrics/collectors/versions/versions_collector.py:20]: Thanks

File: metrics/collectors/versions/versions_collector.py (hunk 1 above)
Comment (seb128) [reply metrics/collectors/versions/versions_collector.py:19]: Changed

File: metrics/collectors/versions/versions_collector.py (hunk 3 above)
Comment (seb128) [reply metrics/collectors/versions/versions_collector.py:28]: changed

**Comment by iainlane** [comment]
I guess said a bit differently - I think that for the desktop team, where we do have an exception, we should plan to track the status of our SRUs as a metric at some point. Not for everything, that's not valuable, but for the desktop team it would be.

Thanks for the updates!

**Comment by seb128** [comment]
Bah, my black local configuration doesn't have the config tweaks the project has. Hold on for the review, I need to fix that and also do the read change you suggested to the bottom part of the function

File: metrics/collectors/versions/versions_collector.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                        continue
+                    category, value = category.split("=")
+                    data.append(
+                        {
+                            "measurement": "versions_script_stats",
+                            "fields": {"count": value},
+                            "tags": {
+                                "report": report,
+                                "category": category,
+                                "serie": "devel",
```
Comment (iainlane) [inline metrics/collectors/versions/versions_collector.py:43]:
```suggestion
                                "series": "devel",
```

:grimacing:

**Comment by seb128** [comment]
Now it seems ready for another review round, the url is using 'current' now which is always available

**Review by iainlane — APPROVED** [review]

**Comment by iainlane** [comment]
Thanks!

**Pull request merged**